FLT/Deformations/Algebra/InverseLimit/Topology.lean

```diff
@@ -0,0 +1,111 @@
+import Mathlib.Algebra.Module.LinearMap.Defs
+import Mathlib.Data.Rat.Cast.Defs
+import Mathlib.Order.DirectedInverseSystem
+import Mathlib.Tactic.SuppressCompilation
+import Mathlib.Order.CompletePartialOrder
+import Mathlib.Order.Defs.Unbundled
+import Mathlib.Algebra.Module.Submodule.Defs
+import Mathlib.Algebra.Module.Pi
+import FLT.Deformations.ContinuousRepresentation.IsTopologicalModule
+import FLT.Deformations.Algebra.InverseLimit.Basic
+
+open TopologicalSpace
+
+variable {ι : Type*} [Preorder ι] {G : ι → Type*}
+variable {T : ∀ ⦃i j : ι⦄, i ≤ j → Type*} {f : ∀ _ _ h, T h}
+variable [∀ i j (h : i ≤ j), FunLike (T h) (G j) (G i)] [InverseSystem (f · · ·)]
+variable [IsDirected ι (· ≤ ·)]
+variable [∀ i : ι, TopologicalSpace (G i)]
+  {cont : ∀ {i j}, (h : i ≤ j) → Continuous (f i j h)}
+
+namespace InverseLimit
+
+variable {W : Type*} {M : ι → Type*} (maps : ∀ i, M i) [∀ i, FunLike (M i) W (G i)]
+variable (inverseSystemHom : InverseSystemHom G f maps)
+variable [TopologicalSpace W]
+variable (maps_cont : (i : ι) → Continuous (maps i))
+
+instance : TopologicalSpace (InverseLimit G f) := by
+  unfold InverseLimit
+  infer_instance
+
+omit [InverseSystem fun x1 x2 x3 ↦ ⇑(f x1 x2 x3)] [IsDirected ι fun x1 x2 ↦ x1 ≤ x2] in
```

Comment:
You might want to do the same experiment on this file. Do you really need these assumptions? If you only need them in a few places, is it easier to add them rather than omit them?